src/common/search/hooks/useSearchFilter.js

```diff
@@ -1,49 +1,53 @@
 import { SearchContext } from "common/search/search-context";
-import { getPlaysOnSearch } from "meta/play-meta-util";
 import { useContext, useEffect, useState } from "react";
+import { useSelector } from "react-redux";
 
 const useSearchFilter = () => {
   const { searchTerm, filterQuery } = useContext(SearchContext);
-  
-  console.log('searchTerm', searchTerm);
-  console.log('filterQuery', filterQuery);
-  
+  const data = useSelector((state) => state?.PlayReducer);
   const [plays, setPlays] = useState([]);
 
   useEffect(() => {
-    const filteredPlays = filterPlays(searchTerm, filterQuery);
+    const filteredPlays = filterPlays(data, searchTerm, filterQuery);
     setPlays(filteredPlays);
-  }, [searchTerm, filterQuery]);
-
+  }, [searchTerm, filterQuery, data]);
+  
   return plays;
 };
 
-const filterPlays = (searchTerm, filterQuery) => {
+const filterPlays = (data, searchTerm, filterQuery) => {
```

Comment:
Check the `TODO 1:` comment. I feel we should pass these filter and search criteria to the hook and let the filtering happen at the server side.

@Angryman18

Reply:
we might wanna send async request to server on keypress delay? the request sends the search term, language and other information based on what backend sends us the component name?
is this what we wanted?

Reply:
Yes, that's right